pkg/scaffold/resource/resource.go

```diff
@@ -78,33 +81,29 @@ func (r *Resource) Validate() error {
 		return fmt.Errorf("kind must be PascalCase (expected %s was %s)", flect.Pascalize(r.Kind), r.Kind)
 	}
 
-	// todo: move it for the proper place since they are not validations and then, should not be here
-	// Add in r.Resource the Kind plural
-	if len(r.Resource) == 0 {
-		r.Resource = flect.Pluralize(strings.ToLower(r.Kind))
-	}
-	// Replace the caracter "-" for "" to allow scaffold the go imports
-	r.GroupImportSafe = strings.Replace(r.Group, "-", "", -1)
-    r.GroupImportSafe = strings.Replace(r.GroupImportSafe, ".", "", -1)
 	return nil
 }
 
-// isKindEmpty will return true if the --kind flag do not be informed
-// NOTE: required check if the flags are assuming the other flags as value
-func (r *Resource) isKindEmpty() bool {
-	return len(r.Kind) == 0 || r.Kind == "--group" || r.Kind == "--version"
+// isEmpty will return true if the provided flag is not set or the value is missing.
+// If provided without value it may take another flag as value, so we need to
+// check if it starts by a dash.
+func (_ *Resource) isEmpty(value string) bool {
```

Comment:
Note that you can remove (_ *Resource) and can the func drectly. Has no need to use `_` here.

Reply:
This is how you implement static method in Golang. A package level function would also work. Compilers are intelligent enough to handle both cases the same way and by using a static method we explicitly show that this method belongs to the `Resource` struct.

Reply:
The Resouce is used when we want to Embedded. In this case, after you change it is a func which has no longer this need.  See: https://golang.org/doc/effective_go.html#embedding

Reply:
You did not understand me. `func (_ *Resource) isEmpty(value string) bool` is equivalent to `func isEmpty(value string) bool` but making it clearer for future readers that this method is used for the `Resource` type. It doesn't have to do with embedding.

@mengqiy @droot Is there any coding policy against static methods in kubebuilder?

Reply:
The [embedding](https://golang.org/doc/effective_go.html#embedding) is for we use to spef that that func is related to the structure. 

For example person.getNames() or person.addAdress(street). In this case, you change the code to a func to check the values and you are passing the values as arg they no longer came from Resources so has no need for it.

Reply:
Thats not embedding. Read the link you provided and you will see that. Embedding is using a field of a struct without a name so that the methods of the embedded type can be used from the wrapper instance without needing to access the field.

What you are trying to say is that receivers are used to avoid passing that variable as a parameter to a function. That is true. Static methods are equivalent to functions in the sense that they don't have a receiver (well, they actually have an empty receiver). It is an OOP concept.

So the short answer is, the blank receiver are not needed. But they are a good way to show the relation between the function and the type. The compiler will handle them the same way, but when reading the code, you inmediately know that it relates to that type.

I already mentioned a couple of the main mantainers of this project. If they do not want to have static methods I will remove the receiver.

Reply:
Let's see if the following comment can clarify:

Here you have 2 options; keep the embedding and not pass the values as arg since they are embedded already or pass as arg as you did but then make no sense keep the embedding`(_ *Resource)`.

Reply:
Again, that is not called embedding, that is called a receiver. And again, using blank receivers is a OOP approach. I will try to explain you what a static method is in Slack, this is not the place to explain what a static method is.